src/system.jl

```diff
@@ -0,0 +1,78 @@
+"""
+    AbstractSystem
+
+Base abstract type for Hamiltonian systems with energy
+    H(q, p) = H₁(q) + H₂(q, p)
+where
+    q   --  current position
+    p   --  current momentum
+    H₁  --  potential energy
+    H₂  --  kinetric energy
```

Comment:
```suggestion
    H₁  --  potential energy
    H₂  --  kinetic energy
```

Though see later discussion around whether we want to draw explicit link between $H_1$ and $H_2$ and potential and kinetic energy terms.